src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -462,6 +509,30 @@ internal void ParseSolution()
                 }
             }
 
+            if (_solutionFilter != null)
+            {
+                HashSet<string> projectPaths = new HashSet<string>(_projectsInOrder.Count, NativeMethodsShared.OSUsesCaseSensitivePaths ? StringComparer.Ordinal : StringComparer.OrdinalIgnoreCase);
+                foreach (ProjectInSolution project in _projectsInOrder)
+                {
+                    projectPaths.Add(project.RelativePath);
+                }
+                foreach (string project in _solutionFilter)
+                {
+                    if (!projectPaths.Contains(project))
+                    {
+                        ProjectFileErrorUtilities.VerifyThrowInvalidProjectFile
+                        (
```

Comment:
is this missing the first parameter (false)?

Reply:
oops, no idea how I ended up commenting on an old PR :)

Reply:
This theoretically will still go in if System.Text.Json starts working. This PR also specifies what VerifyThrowInvalidProjectFile should do if it's missing the first parameter: assume false.

Reply:
Ah - I thought the overloads that assume false were named `ThrowInvalidProjectFile()`